### ISSUE TITLE
Linux 32 bit build issue

### ISSUE THREAD

**Issue by nsauzede** [franzflasch/linux_for_riscv_em#1]
I seem to have an issue building the Linux 32 bit payload on Ubuntu 21.10:
```
nico@lenovo:~/perso/git/linux_for_riscv_em$ ./riscv_em_rv32_mmu_linux.sh  
...
>>> host-m4 1.4.18 Building
PATH="/home/nico/perso/git/linux_for_riscv_em/output_mmu_rv32/buildroot-2021.02.1/output/host/bin:/home/nico/perso/git/linux_for_riscv_em/output_mmu_rv32/buildroot-2021.02.1/output/host/sbin:~/bin:/home/nico/Downloads/clang+llvm-12.0.0-x86_64-linux-gnu-ubuntu-20.04/bin:~/bin:/home/nico/Downloads/clang+llvm-12.0.0-x86_64-linux-gnu-ubuntu-20.04/bin:/home/nico/.cargo/bin:/home/nico/.local/bin:/home/nico/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/nico/perso/git/riscv-gnu-toolchain/the_install/bin:/home/nico/perso/git/vade:/home/nico/perso/git/riscv-gnu-toolchain/the_install/bin" PKG_CONFIG="/home/nico/perso/git/linux_for_riscv_em/output_mmu_rv32/buildroot-2021.02.1/output/host/bin/pkg-config" PKG_CONFIG_SYSROOT_DIR="/" PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 PKG_CONFIG_ALLOW_SYSTEM_LIBS=1 PKG_CONFIG_LIBDIR="/home/nico/perso/git/linux_for_riscv_em/output_mmu_rv32/buildroot-2021.02.1/output/host/lib/pkgconfig:/home/nico/perso/git/linux_for_riscv_em/output_mmu_rv32/buildroot-2021.02.1/output/host/share/pkgconfig"  /usr/bin/make   -C /home/nico/perso/git/linux_for_riscv_em/output_mmu_rv32/buildroot-2021.02.1/output/build/host-m4-1.4.18/
/usr/bin/make  all-recursive
Making all in .
make[4]: Nothing to be done for 'all-am'.
Making all in examples
make[4]: Nothing to be done for 'all'.
Making all in lib
  GEN      alloca.h
  GEN      configmake.h
  GEN      c++defs.h
  GEN      arg-nonnull.h
  GEN      warn-on-use.h
  GEN      unused-parameter.h
  GEN      limits.h
  GEN      sys/types.h
  GEN      fcntl.h
  GEN      locale.h
  GEN      langinfo.h
  GEN      math.h
  GEN      signal.h
  GEN      spawn.h
  GEN      stdio.h
  GEN      stdlib.h
  GEN      string.h
  GEN      sys/stat.h
  GEN      sys/time.h
  GEN      sys/wait.h
  GEN      time.h
  GEN      unistd.h
  GEN      wchar.h
  GEN      wctype.h
/usr/bin/make  all-am
  CC       binary-io.o
  CC       gl_avltree_oset.o
  CC       c-ctype.o
  CC       c-stack.o
  CC       c-strcasecmp.o
  CC       c-strncasecmp.o
  CC       clean-temp.o
  CC       cloexec.o
  CC       close-stream.o
  CC       closein.o
  CC       closeout.o
  CC       dirname.o
  CC       basename.o
  CC       dirname-lgpl.o
  CC       basename-lgpl.o
  CC       stripslash.o
  CC       execute.o
  CC       exitfail.o
In file included from /usr/include/signal.h:328,
                 from ./signal.h:52,
                 from c-stack.c:49:
c-stack.c:55:26: error: missing binary operator before token "("
   55 | #elif HAVE_LIBSIGSEGV && SIGSTKSZ < 16384
      |                          ^~~~~~~~
  CC       fatal-signal.o
  CC       fd-hook.o
make[5]: *** [Makefile:1915: c-stack.o] Error 1
make[5]: *** Waiting for unfinished jobs....
  CC       fd-safer-flag.o
make[4]: *** [Makefile:1674: all] Error 2
make[3]: *** [Makefile:1572: all-recursive] Error 1
make[2]: *** [Makefile:1528: all] Error 2
make[1]: *** [package/pkg-generic.mk:250: /home/nico/perso/git/linux_for_riscv_em/output_mmu_rv32/buildroot-2021.02.1/output/build/host-m4-1.4.18/.stamp_built] Error 2
make: *** [Makefile:84: _all] Error 2
```
Do you have an idea of what's happening or something I missed ?
Note that I didn't follow your first step as I already had a riscv32 (rv32g) toolchain installed according to this doc:
https://github.com/nsauzede/myrv
Maybe the error actually comes from the alternate riscv toolchain ?

**Comment by franzflasch** [comment]
Seems that this is a buildroot issue, not related to the toolchain: 
http://lists.busybox.net/pipermail/buildroot/2021-April/308176.html

Probably only occurs with newer Ubuntu releases like yours. (I've tested it with Ubuntu 20.04)

I will test it myself with Ubuntu 21.10 tomorrow and report you back.

**Comment by franzflasch** [comment]
Should be fixed with 18d413df36ed7ccd83e6cbe9fd1c8d11dc9fb3a4

As initially suspected it was a buildroot issue, which is already fixed in newer versions. So I've updated to buildroot 2021.02.6 which solved it. (At least for me)

Please check if it also works for you now.

**Comment by nsauzede** [comment]
Yes! I confirm your commit fixes my issue. Thank you very much.
(I also had to install your actual Riscv cross compiler, as mine was not the correct one (targeting bare metal instead of linux)
